## Importing Data and Visualization of Entry Format

Import pandas and numpy

In [1]:
import pandas as pd
import numpy as np

Import data

In [2]:
autos = pd.read_csv("autos.csv", encoding = "Latin-1")

Visualize entries and format

In [3]:
autos

dateCrawled                                          name  \
0       2016-03-24 11:52:17                                    Golf_3_1.6   
1       2016-03-24 10:58:45                          A5_Sportback_2.7_Tdi   
2       2016-03-14 12:52:21                Jeep_Grand_Cherokee_"Overland"   
3       2016-03-17 16:54:04                            GOLF_4_1_4__3TÜRER   
4       2016-03-31 17:25:20                Skoda_Fabia_1.4_TDI_PD_Classic   
...                     ...                                           ...   
371523  2016-03-14 17:48:27                    Suche_t4___vito_ab_6_sitze   
371524  2016-03-05 19:56:21         Smart_smart_leistungssteigerung_100ps   
371525  2016-03-19 18:57:12            Volkswagen_Multivan_T4_TDI_7DC_UY2   
371526  2016-03-20 19:41:08                        VW_Golf_Kombi_1_9l_TDI   
371527  2016-03-07 19:39:19  BMW_M135i_vollausgestattet_NP_52.720____Euro   

        seller offerType  price   abtest vehicleType  yearOfRegistration  \
0       privat   Angebot    480     test         NaN                1993   
1       privat   Angebot  18300     test       coupe                2011   
2       privat   Angebot   9800     test         suv                2004   
3       privat   Angebot   1500     test  kleinwagen                2001   
4       privat   Angebot   3600     test  kleinwagen                2008   
...        ...       ...    ...      ...         ...                 ...   
371523  privat   Angebot   2200     test         NaN                2005   
371524  privat   Angebot   1199     test      cabrio                2000   
371525  privat   Angebot   9200     test         bus                1996   
371526  privat   Angebot   3400     test       kombi                2002   
371527  privat   Angebot  28990  control   limousine                2013   

          gearbox  powerPS        model  kilometer  monthOfRegistration  \
0         manuell        0         golf     150000                    0   
1         manuell      190          NaN     125000                    5   
2       automatik      163        grand     125000                    8   
3         manuell       75         golf     150000                    6   
4         manuell       69        fabia      90000                    7   
...           ...      ...          ...        ...                  ...   
371523        NaN        0          NaN      20000                    1   
371524  automatik      101       fortwo     125000                    3   
371525    manuell      102  transporter     150000                    3   
371526    manuell      100         golf     150000                    6   
371527    manuell      320      m_reihe      50000                    8   

       fuelType           brand notRepairedDamage          dateCreated  \
0        benzin      volkswagen               NaN  2016-03-24 00:00:00   
1        diesel            audi                ja  2016-03-24 00:00:00   
2        diesel            jeep               NaN  2016-03-14 00:00:00   
3        benzin      volkswagen              nein  2016-03-17 00:00:00   
4        diesel           skoda              nein  2016-03-31 00:00:00   
...         ...             ...               ...                  ...   
371523      NaN  sonstige_autos               NaN  2016-03-14 00:00:00   
371524   benzin           smart              nein  2016-03-05 00:00:00   
371525   diesel      volkswagen              nein  2016-03-19 00:00:00   
371526   diesel      volkswagen               NaN  2016-03-20 00:00:00   
371527   benzin             bmw              nein  2016-03-07 00:00:00   

        nrOfPictures  postalCode             lastSeen  
0                  0       70435  2016-04-07 03:16:57  
1                  0       66954  2016-04-07 01:46:50  
2                  0       90480  2016-04-05 12:47:46  
3                  0       91074  2016-03-17 17:40:17  
4                  0       60437  2016-04-06 10:17:21  
...              ...         ...                  ..

In [4]:
print(autos.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

## Column Name Cleaning

Print existing column names

In [34]:
autos.columns

Index(['Date_Crawled', 'Name', 'Price', 'AB_Test', 'Vehicle_Type',
       'Registration_Year', 'Transmission', 'PS', 'Model', 'Odometer',
       'Registration_Month', 'Fuel_Type', 'Brand', 'Damage', 'Listing_Date',
       'Postal_Code', 'Last_Online'],
      dtype='object')

Change column names to a more readable format

In [6]:
autos.columns = ['Date_Crawled', 'Name', 'Seller', 'Offer_Type', 'Price', 'AB_Test', 'Vehicle_Type', 'Registration_Year', 'Transmission', 'PS', 'Model',
       'Odometer', 'Registration_Month', 'Fuel_Type', 'Brand', 'Damage', 'Listing_Date', 'Photo_Qty', 'Postal_Code', 'Last_Online']

autos.head()

Date_Crawled                            Name  Seller Offer_Type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   Price AB_Test Vehicle_Type  Registration_Year Transmission   PS  Model  \
0    480    test          NaN               1993      manuell    0   golf   
1  18300    test        coupe               2011      manuell  190    NaN   
2   9800    test          suv               2004    automatik  163  grand   
3   1500    test   kleinwagen               2001      manuell   75   golf   
4   3600    test   kleinwagen               2008      manuell   69  fabia   

   Odometer  Registration_Month Fuel_Type       Brand Damage  \
0    150000                   0    benzin  volkswagen    NaN   
1    125000                   5    diesel        audi     ja   
2    125000                   8    diesel        jeep    NaN   
3    150000                   6    benzin  volkswagen   nein   
4     90000                   7    diesel       skoda   nein   

          Listing_Date  Photo_Qty  Postal_Code          Last_Online  
0  2016-03-24 00:00:00          0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00          0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00          0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00          0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00          0        60437  2016-04-06 10:17:21

## Data Exploration

Show entry information

In [7]:
autos.describe(include = 'all')

Date_Crawled         Name  Seller Offer_Type         Price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

       AB_Test Vehicle_Type  Registration_Year Transmission             PS  \
count   371528       333659      371528.000000       351319  371528.000000   
unique       2            8                NaN            2            NaN   
top       test    limousine                NaN      manuell            NaN   
freq    192585        95894                NaN       274214            NaN   
mean       NaN          NaN        2004.577997          NaN     115.549477   
std        NaN          NaN          92.866598          NaN     192.139578   
min        NaN          NaN        1000.000000          NaN       0.000000   
25%        NaN          NaN        1999.000000          NaN      70.000000   
50%        NaN          NaN        2003.000000          NaN     105.000000   
75%        NaN          NaN        2008.000000          NaN     150.000000   
max        NaN          NaN        9999.000000          NaN   20000.000000   

         Model       Odometer  Registration_Month Fuel_Type       Brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

        Damage         Listing_Date  Photo_Qty   Postal_Code  \
count   299468               371528   371528.0  371528.00000   
unique       2                  114        NaN           NaN   
top       nein  2016-04-03 00:00:00        NaN           NaN   
freq    263182                14450        NaN           NaN   
mean       NaN                  NaN        0.0   50820.66764   
std        NaN                  NaN        0.0   25799.08247   
min        NaN                  NaN        0.0    1067.00000   
25%        NaN                  NaN        0.0   30459.00000   
50%        NaN                  NaN        0.0   49610.00000   
75%        NaN                  NaN        0.0   71546.00000   
max        NaN                  NaN        0.0   99998.00000   

                Last_Online  
count                371528  
unique               182806  
top     2016-04-06 13:45:54  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

Explore why Photo_Qty has 0 values for statistics

In [8]:
autos["Photo_Qty"].value_counts()

Photo_Qty
0    371528
Name: count, dtype: int64

Remove all unnecessary cols

In [9]:
autos = autos.drop(["Photo_Qty", "Seller", "Offer_Type"], axis=1)

Clean and convert special characters in price column. Also add English translations for German words.

In [10]:
# Ensure price values are strings
autos["Price"] = autos["Price"].astype(str)

# convert old to new
autos["Price"] = (autos["Price"]
                  .str.replace("$", "", regex=False)
                  .str.replace(",", "", regex=False)
                  .astype(int)
                  )

autos["Vehicle_Type"] = (autos["Vehicle_Type"]
                  .str.replace("kleinwagen", "hatchback", regex=False)
                  .str.replace("limousine", "sedan", regex=False)
                  .str.replace("cabrio", "convertible", regex=False)
                  .str.replace("kombi", "wagon", regex=False)
                  )

autos["Transmission"] = (autos["Transmission"]
                  .str.replace("manuell", "manual", regex=False)
                  .str.replace("automatik", "automatic", regex=False)
                  )

autos["Fuel_Type"] = (autos["Fuel_Type"]
                  .str.replace("benzin", "gas", regex=False)
                 )

autos["Damage"] = (autos["Damage"]
                  .str.replace("nein", "no", regex=False)
                  .str.replace("ja", "yes", regex=False)
                 )

autos.head()

Date_Crawled                            Name  Price AB_Test  \
0  2016-03-24 11:52:17                      Golf_3_1.6    480    test   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  18300    test   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"   9800    test   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER   1500    test   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic   3600    test   

  Vehicle_Type  Registration_Year Transmission   PS  Model  Odometer  \
0          NaN               1993       manual    0   golf    150000   
1        coupe               2011       manual  190    NaN    125000   
2          suv               2004    automatic  163  grand    125000   
3    hatchback               2001       manual   75   golf    150000   
4    hatchback               2008       manual   69  fabia     90000   

   Registration_Month Fuel_Type       Brand Damage         Listing_Date  \
0                   0       gas  volkswagen    NaN  2016-03-24 00:00:00   
1                   5    diesel        audi    yes  2016-03-24 00:00:00   
2                   8    diesel        jeep    NaN  2016-03-14 00:00:00   
3                   6       gas  volkswagen     no  2016-03-17 00:00:00   
4                   7    diesel       skoda     no  2016-03-31 00:00:00   

   Postal_Code          Last_Online  
0        70435  2016-04-07 03:16:57  
1        66954  2016-04-07 01:46:50  
2        90480  2016-04-05 12:47:46  
3        91074  2016-03-17 17:40:17  
4        60437  2016-04-06 10:17:21

## Exploring Odometer/Price

Visualize data to look for common values

In [11]:
autos["Odometer"].value_counts()

Odometer
150000    240797
125000     38067
100000     15920
90000      12523
80000      11053
70000       9773
60000       8669
50000       7615
5000        7069
40000       6376
30000       6041
20000       5676
10000       1949
Name: count, dtype: int64

Display repetitive values

In [12]:
print(autos["Price"].unique().shape) # gives number of unique values in the Price field
print(autos["Price"].describe())
autos["Price"].value_counts().head(20)

(5597,)
count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: Price, dtype: float64


Price
0       10778
500      5670
1500     5394
1000     4649
1200     4594
2500     4438
600      3819
3500     3792
800      3784
2000     3432
999      3364
750      3203
650      3150
4500     3053
850      2946
700      2936
2200     2936
1800     2886
900      2874
950      2793
Name: count, dtype: int64

Print the high values to see where prices jump to outlier levels. (Jump seen after 3895000).

In [13]:
autos["Price"].value_counts().sort_index(ascending = False).head(20)

Price
2147483647     1
99999999      15
99000000       1
74185296       1
32545461       1
27322222       1
14000500       1
12345678       9
11111111      10
10010011       1
10000000       8
9999999        3
3895000        1
3890000        1
2995000        1
2795000        1
1600000        2
1300000        1
1250000        2
1234566        1
Name: count, dtype: int64

Print low values. We cannot eliminate values aside from 0 as $1 is a legitimate bid on ebay.

In [14]:
autos["Price"].value_counts().sort_index(ascending = True).head(20)

Price
0     10778
1      1189
2        12
3         8
4         1
5        26
7         3
8         9
9         8
10       84
11        5
12        8
13        7
14        5
15       27
16        2
17        5
18        3
19        3
20       51
Name: count, dtype: int64

Cut out outlier listings and zero price listings

In [15]:
autos = autos[autos["Price"].between(1, 3895001)] # keep rows with price between these values
autos["Price"].describe()

count    3.607340e+05
mean     7.701370e+03
std      2.425602e+05
min      1.000000e+00
25%      1.250000e+03
50%      3.000000e+03
75%      7.490000e+03
max      9.900000e+07
Name: Price, dtype: float64

## Exploring the Date Column

In [16]:
# structure visualization
autos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 360734 entries, 0 to 371527
Data columns (total 17 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   Date_Crawled        360734 non-null  object
 1   Name                360734 non-null  object
 2   Price               360734 non-null  int64 
 3   AB_Test             360734 non-null  object
 4   Vehicle_Type        326611 non-null  object
 5   Registration_Year   360734 non-null  int64 
 6   Transmission        343011 non-null  object
 7   PS                  360734 non-null  int64 
 8   Model               342437 non-null  object
 9   Odometer            360734 non-null  int64 
 10  Registration_Month  360734 non-null  int64 
 11  Fuel_Type           330792 non-null  object
 12  Brand               360734 non-null  object
 13  Damage              293975 non-null  object
 14  Listing_Date        360734 non-null  object
 15  Postal_Code         360734 non-null  int64 
 16  Last_On

Visualize date entries

In [17]:
autos[["Date_Crawled", "Listing_Date", "Last_Online"]].head()

Date_Crawled         Listing_Date          Last_Online
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

Only need date, keep only the first 10 characters

In [19]:
# date frequency

(autos["Date_Crawled"]
         .str[:10]
         .value_counts(normalize = True, dropna = False)
         .sort_index()
         )

Date_Crawled
2016-03-05    0.025545
2016-03-06    0.014479
2016-03-07    0.035650
2016-03-08    0.033473
2016-03-09    0.034114
2016-03-10    0.032645
2016-03-11    0.032775
2016-03-12    0.036243
2016-03-13    0.015779
2016-03-14    0.036331
2016-03-15    0.033421
2016-03-16    0.030200
2016-03-17    0.031652
2016-03-18    0.013118
2016-03-19    0.035270
2016-03-20    0.036395
2016-03-21    0.035697
2016-03-22    0.032489
2016-03-23    0.032001
2016-03-24    0.029911
2016-03-25    0.032797
2016-03-26    0.031974
2016-03-27    0.030227
2016-03-28    0.035059
2016-03-29    0.034122
2016-03-30    0.033532
2016-03-31    0.031879
2016-04-01    0.034153
2016-04-02    0.035095
2016-04-03    0.038818
2016-04-04    0.037626
2016-04-05    0.012785
2016-04-06    0.003127
2016-04-07    0.001619
Name: proportion, dtype: float64

In [20]:
(autos["Date_Crawled"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_values()
        )

Date_Crawled
2016-04-07    0.001619
2016-04-06    0.003127
2016-04-05    0.012785
2016-03-18    0.013118
2016-03-06    0.014479
2016-03-13    0.015779
2016-03-05    0.025545
2016-03-24    0.029911
2016-03-16    0.030200
2016-03-27    0.030227
2016-03-17    0.031652
2016-03-31    0.031879
2016-03-26    0.031974
2016-03-23    0.032001
2016-03-22    0.032489
2016-03-10    0.032645
2016-03-11    0.032775
2016-03-25    0.032797
2016-03-15    0.033421
2016-03-08    0.033473
2016-03-30    0.033532
2016-03-09    0.034114
2016-03-29    0.034122
2016-04-01    0.034153
2016-03-28    0.035059
2016-04-02    0.035095
2016-03-19    0.035270
2016-03-07    0.035650
2016-03-21    0.035697
2016-03-12    0.036243
2016-03-14    0.036331
2016-03-20    0.036395
2016-04-04    0.037626
2016-04-03    0.038818
Name: proportion, dtype: float64

In [21]:
(autos["Last_Online"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        )

Last_Online
2016-03-05    0.001267
2016-03-06    0.004097
2016-03-07    0.005201
2016-03-08    0.007942
2016-03-09    0.009824
2016-03-10    0.011460
2016-03-11    0.012954
2016-03-12    0.023242
2016-03-13    0.008408
2016-03-14    0.012172
2016-03-15    0.016325
2016-03-16    0.016414
2016-03-17    0.028705
2016-03-18    0.006892
2016-03-19    0.016331
2016-03-20    0.019887
2016-03-21    0.020031
2016-03-22    0.020508
2016-03-23    0.018016
2016-03-24    0.019161
2016-03-25    0.019000
2016-03-26    0.015956
2016-03-27    0.016719
2016-03-28    0.022191
2016-03-29    0.023283
2016-03-30    0.023724
2016-03-31    0.024239
2016-04-01    0.023901
2016-04-02    0.024974
2016-04-03    0.025321
2016-04-04    0.025529
2016-04-05    0.126966
2016-04-06    0.218940
2016-04-07    0.130420
Name: proportion, dtype: float64

Given that the last 3 days have high numbers of last seen, we can assume most of the listings from those days were only posted and not finalized

In [22]:
print(autos["Listing_Date"].str[:10].unique().shape)
(autos["Listing_Date"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        )

(114,)


Listing_Date
2014-03-10    0.000003
2015-03-20    0.000003
2015-06-11    0.000003
2015-06-18    0.000003
2015-08-07    0.000003
                ...   
2016-04-03    0.039007
2016-04-04    0.037734
2016-04-05    0.011618
2016-04-06    0.003119
2016-04-07    0.001555
Name: proportion, Length: 114, dtype: float64

There is a large variety of ad created dates. Most fall within 1-2 months of the listing date, with the oldest at around 9 months.

In [23]:
autos["Registration_Year"].describe()

count    360734.000000
mean       2004.450956
std          82.106279
min        1000.000000
25%        1999.000000
50%        2004.000000
75%        2008.000000
max        9999.000000
Name: Registration_Year, dtype: float64

The minimum value is 1000 and the maximum is 9999, therefore this data will need to be cleaned

## Incorrect Registration Year Data

Display distribution of listings that fall outside of relistic registration year data

In [24]:
(~autos["Registration_Year"].between(1900, 2016)).sum() / autos.shape[0]

np.float64(0.03876263396297549)

Remove the data as it only makes up a small fraction of our data

In [25]:
autos = autos[autos["Registration_Year"].between(1900, 2016)]
autos["Registration_Year"].value_counts(normalize=True).head(10)

Registration_Year
2000    0.066711
1999    0.063538
2005    0.062665
2006    0.057701
2001    0.056949
2003    0.056551
2004    0.056167
2002    0.054278
2007    0.050489
1998    0.049687
Name: proportion, dtype: float64

Most of the vehicles were first registered in the past 20 years

## Exploring Price by Band

Display the unique values in the brand column

In [26]:
autos["Brand"].value_counts(normalize=True)

Brand
volkswagen        0.211691
bmw               0.109868
opel              0.106390
mercedes_benz     0.096827
audi              0.089540
ford              0.068911
renault           0.047504
peugeot           0.030146
fiat              0.025687
seat              0.018656
skoda             0.015683
mazda             0.015383
smart             0.014327
citroen           0.013949
nissan            0.013595
toyota            0.012929
hyundai           0.009970
sonstige_autos    0.009552
mini              0.009381
volvo             0.009145
mitsubishi        0.008234
honda             0.007530
kia               0.006913
suzuki            0.006365
alfa_romeo        0.006310
porsche           0.006261
chevrolet         0.005024
chrysler          0.003862
dacia             0.002495
jeep              0.002195
land_rover        0.002166
daihatsu          0.002160
subaru            0.002117
jaguar            0.001736
saab              0.001465
daewoo            0.001456
trabant           0.00

Since the top six brands have over 5% of the data respectively, we will highlight those brands

In [27]:
brand_count = autos["Brand"].value_counts(normalize = True)
common = brand_count[brand_count > 0.05].index 
print(common)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object', name='Brand')


Show average price of the top 6 brands

In [28]:
brand_mean_p = {}

for Brand in common:
    brand_only = autos[autos["Brand"] == Brand]
    Mean_p = brand_only["Price"].mean()
    brand_mean_p[Brand] = int(Mean_p)

brand_mean_p

{'volkswagen': 7272,
 'bmw': 9952,
 'opel': 3363,
 'mercedes_benz': 9025,
 'audi': 13431,
 'ford': 4761}

## Exploring Mileage

Pandas series constructor and pandas dataframe constructor to compare average mileage and average price of top 6 brands

In [29]:
bmp_ser = pd.Series(brand_mean_p)
bmp_ser

volkswagen        7272
bmw               9952
opel              3363
mercedes_benz     9025
audi             13431
ford              4761
dtype: int64

In [30]:
df = pd.DataFrame(bmp_ser, columns = ["Mean_Price"])
df

Mean_Price
volkswagen           7272
bmw                  9952
opel                 3363
mercedes_benz        9025
audi                13431
ford                 4761

In [31]:
# find mean mileage and mean price for top brands and store in a dictionary

brand_mean_m = {}

for Brand in common:
    brand_only = autos[autos["Brand"] == Brand]
    Mean_m = brand_only["Odometer"].mean()
    brand_mean_m[Brand] = int(Mean_m)

Mean_m = pd.Series(brand_mean_m).sort_values(ascending = False)
mean_prices = pd.Series(brand_mean_p).sort_values(ascending=False)

In [32]:
brand_info = pd.DataFrame(Mean_m,columns=['Mean_Mileage'])
brand_info

Mean_Mileage
bmw                  132789
mercedes_benz        130566
audi                 129440
opel                 128721
volkswagen           128387
ford                 123661

In [33]:
brand_info["Mean_Price"] = mean_prices
brand_info

Mean_Mileage  Mean_Price
bmw                  132789        9952
mercedes_benz        130566        9025
audi                 129440       13431
opel                 128721        3363
volkswagen           128387        7272
ford                 123661        4761